src/mesh/coordinates.cxx

```diff
@@ -1242,9 +1242,10 @@ int Coordinates::calcCovariant(const std::string& region) {
     a(1, 2) = a(2, 1) = g23[i];
     a(0, 2) = a(2, 0) = g13[i];
 
-    if (!invert3x3(a)) {
-      output_error.write("\tERROR: metric tensor is singular at ({:d}, {:d})\n", i.x(),
-                         i.y());
+    if (const auto det = bout::invert3x3(a); det.has_value()) {
+      output_error.write(
+          "\tERROR: metric tensor is singular at ({:d}, {:d}), determinant: {:d}\n",
+          i.x(), i.y(), det.value());
```

Comment:
This could be 3D compatible:
```suggestion
          "\tERROR: metric tensor is singular at {}, determinant: {:d}\n",
          i, det.value());
```

Reply:
Nice, good catch, thanks!